File: 1716.calculate-money-in-leetcode-bank.cpp
/*
 * @lc app=leetcode id=1716 lang=cpp
 *
 * [1716] Calculate Money in Leetcode Bank
 */

// @lc code=start
class Solution {
public:
    int totalMoney(int n) {
        int weeks = n / 7;
        int days = n % 7;
        int result = 0;

        for (int i = 0; i < weeks; i++) {
            result += 28 + i * 7;
        }

        for (int i = 0; i < days; i++) {
            result += (i + 1) + weeks;
        }

        return result;
    }
};

// @lc code=end

File: 1716.calculate-money-in-leetcode-bank.py
#
# @lc app=leetcode id=1716 lang=python3
#
# [1716] Calculate Money in Leetcode Bank
#

# @lc code=start
class Solution:
    def totalMoney(self, n: int) -> int:
        weeks, days, result = n // 7, n % 7, 0
        for i in range(weeks):
            result += 28 + i * 7
        for i in range(days):
            result += (i + 1) + weeks
        return result
        
# @lc code=end

